src/lib/crypto/kimchi_backend/gadgets/range_check.ml

```diff
@@ -0,0 +1,296 @@
+open Core_kernel
+
+open Kimchi_backend_common.Plonk_constraint_system.Plonk_constraint
+
+(* Helper to create RangeCheck0 gate, configured in various ways
+ *     - is_64bit   : create 64-bit range check
+ *     - is_compact : compact limbs mode (only used by compact multi-range-check)
+ *)
+let range_check0 (type f)
+    (module Circuit : Snarky_backendless.Snark_intf.Run with type field = f)
```

Comment:
nit: note that in the rest of the codebase we call this module `Impl` instead of `Circuit`. It might be good to stay consistent here